### ISSUE TITLE
Feature Request: solid.auth.register

### ISSUE THREAD

**Issue by boulderwebdev** [solid/solid-auth-client#102]
It would be nice if there was an API call in `solid.auth` similar to `login` where you pass in the URL of an identity provider and you are sent over to their registration page. For example,
```javascript
await solid.auth.register('https://solid.auth.community/')
```
would redirect you to `https://solid.community/register` with the correct query parameters (mainly redirect_uri).

**Comment by rimmartin** [comment]
I second the motion

**Comment by RubenVerborgh** [comment]
Indeed; depends on https://github.com/solid/solid-spec/issues/138 though.

**Comment by timbl** [comment]
Please explain the block.

**Comment by RubenVerborgh** [comment]
Just to make sure we're on the same page:
- this issue currently has a label "blocked"
- that label means "depends on another piece of work being performed first"  (so we are _blocked_ from working on this)

The blocking is explained by my comment above: we first need to implement https://github.com/solid/solid-spec/issues/138, such that there is a standard way of discovering the registration page from the IDP URL. The `/register` convention is non-standard.

If there is already a standard for discovering the registration page, then let's unblock this issue by adding it to the spec and implementing it.

**Comment by RubenVerborgh** [comment]
I renamed the label to "has dependency" to remove the confusion (originally seen at https://gitter.im/solid/specification?at=5f77b16d717f8e4eb5f67494).

**Comment by RubenVerborgh** [comment]
@gibsonf1 kindly [pointed out](https://gitter.im/solid/specification?at=5f78632e66ee4e44a87ce68c) that the necessary API already exists. For instance, https://drive.verborgh.org/.well-known/openid-configuration shows https://drive.verborgh.org/register. Hence, there is no blocker indeed.

**Comment by elf-pavlik** [comment]
https://openid.net/specs/openid-connect-discovery-1_0.html

> registration_endpoint
    RECOMMENDED. URL of the OP's Dynamic Client Registration Endpoint [OpenID.Registration]. 


It is supposed to be used by OAuth2 client applications to dynamically register, not by users to create accounts. For that maybe solid/solid-oidc#77 could come handy.
